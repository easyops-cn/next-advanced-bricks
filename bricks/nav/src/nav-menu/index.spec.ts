import { describe, test, expect } from "@jest/globals";
import { act } from "react-dom/test-utils";
import "./index.js";
import { NavMenu } from "./index.js";

jest.mock("@next-core/theme", () => ({}));

jest.mock("@next-core/runtime", () => ({
  getHistory: () => ({
    location: {},
    listen: () => {
      //
    },
    createHref: () => "/oops",
  }),
  matchPath: () => true,
}));

describe("eo-nav-menu", () => {
  test("basic usage", async () => {
    (window as any).ResizeObserver = jest.fn().mockImplementation(() => ({
      observe: jest.fn(),
      disconnect: jest.fn(),
    }));
    const element = document.createElement("eo-nav-menu") as NavMenu;

    expect(element.shadowRoot).toBeFalsy();
    act(() => {
      element.menu = {
        title: "mock a",
        menuItems: [
          {
            text: "菜单一",
            type: "default",
            key: "menu 1",
          },
          {
            title: "菜单二",
            type: "subMenu",
            key: "menu 2",
            items: [
              {
                text: "菜单二-1",
                type: "default",
                key: "menu 3",
              },
              {
                title: "分组",
                type: "group",
                key: "menu 4",
                items: [
                  {
                    text: "分组1",
                    to: "/oops",
                    type: "default",
                    key: "menu 5",
                  },
                  {
                    text: "分组2",
                    type: "default",
                    key: "menu 6",
                  },
                ],
              },
            ],
          },
          {
            title: "菜单三",
            type: "group",
            childLayout: "siteMap",
            groupFrom: "platformMenus",
            groupId: "platformMenus",
            items: [
              {
                childLayout: "default",

                items: [
                  {
                    text: "MySQL",
                    to: "/database-service-monitor/MySQL",
                    type: "default",
                  },
                ],
                key: "2.0",
                title: "数据库服务",
                type: "group",
              },
            ],
          },
        ],
      };
      document.body.appendChild(element);
    });
    expect(element.shadowRoot).toBeTruthy();
    expect(element.shadowRoot?.childNodes.length).toBe(2);

    expect(element.shadowRoot?.innerHTML).toMatchSnapshot();

    act(() => {
      document.body.removeChild(element);
    });

    expect(document.body.contains(element)).toBeFalsy();
  });
});
