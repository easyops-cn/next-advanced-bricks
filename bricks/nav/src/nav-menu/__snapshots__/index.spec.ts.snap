// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eo-nav-menu basic usage 1`] = `
"<style>SiteMapItem.shadow.css
ItemTag.shadow.css
GroupItem.shadow.css
nav-menu.shadow.css</style><div class="nav-menu-wrapper"><eo-menu-item data-index="0" class="" title=""><eo-link type="plain" style="width: 100%;"><span class="menu-item-label" style="color: red;">菜单一</span></eo-link></eo-menu-item><eo-popover data-index="1" class="popover" trigger="hover" placement="bottom-start" distance="0" anchordisplay="block" strategy="fixed"><eo-menu-item class="sub-menu-item-label 22" slot="anchor" title=""><span class="menu-item-label" style="color: red;">菜单二</span></eo-menu-item><div class="sub-menu-wrapper"><eo-menu-item class="" title=""><eo-link type="plain" style="width: 100%;"><span class="menu-item-label">菜单二-1</span></eo-link></eo-menu-item><div class="group-label">分组</div><div class="group-wrapper"><eo-menu-item class="" title="" active=""><eo-link type="plain" url="/oops" style="width: 100%;"><span class="menu-item-label">分组1</span></eo-link></eo-menu-item><eo-menu-item class="" title=""><eo-link type="plain" style="width: 100%;"><span class="menu-item-label">分组2</span></eo-link></eo-menu-item></div></div></eo-popover><eo-popover data-index="2" class="popover" trigger="hover" placement="bottom-start" distance="0" strategy="fixed"><eo-menu-item class="sub-menu-item-label" slot="anchor" title=""><span class="menu-item-label" style="color: red;">菜单三</span></eo-menu-item><div class="sub-menu-sit-map-wrapper"></div></eo-popover><slot name="navMenuRightContent"></slot></div>"
`;
